src/Serilog.Sinks.RollingFile/Sinks/RollingFile/Specifier.cs

```diff
@@ -0,0 +1,118 @@
+﻿using System;
+
+namespace Serilog.Sinks.RollingFile.Sinks.RollingFile
+{
+    internal class Specifier
+    {
+        internal const string OldStyleDateToken = "{0}";
+
+        internal const string DateToken = "{Date}";
+        internal const string HourToken = "{Hour}";
+        internal const string HalfHourToken = "{HalfHour}";
+
+        internal const string DateFormat = "yyyyMMdd";
+        internal const string HourFormat = "yyyyMMddHH";
+        internal const string HalfHourFormat = "yyyyMMddHHmm";
+
+        internal static readonly TimeSpan DateInterval = TimeSpan.FromDays(1);
+        internal static readonly TimeSpan HourInterval = TimeSpan.FromHours(1);
+        internal static readonly TimeSpan HalfHourInterval = TimeSpan.FromMinutes(30);
+
+        //-------------------
+
+        public static readonly Specifier Date = new Specifier(SpecifierType.Date);
+        public static readonly Specifier Hour = new Specifier(SpecifierType.Hour);
+        public static readonly Specifier HalfHour = new Specifier(SpecifierType.HalfHour);
+
+        //-------------------
+
+
+        public SpecifierType Type { get; }
+        public string Name { get; }
+        public string Token { get; }
+        public string Format { get; }
+        public TimeSpan Interval { get; }
+
+        Specifier(SpecifierType type)
```

Comment:
SpecifierType is used also in the getCurrentCheckpoint function. I'll remove this enum and I'll use the token to identify what kind of specifier is the instance.